languages/thingtalk/load-thingpedia.ts

```diff
@@ -1192,10 +1209,32 @@ export default class ThingpediaLoader {
             }
             this._initialFunction = functionDef;
         }
+
+        const followUp = functionDef.getImplementationAnnotation<string>('follow_up');
+        if (followUp) {
```

Comment:
when I saw the if statement in the annotation, I was like "this is neat, how is it done in thingtalk?", then I saw this... lol
we do plan to generalize it and do it in thingtalk in the future right?

Reply:
The plan is not to use the `#[follow_up]` annotation at all. Instead, if you want custom agent behavior you write a tiny JS or TT function that replaces the agent policy entirely, while the skill is active. The design is still in flux, between me Monica and Chaofei. You can join the discussion on Slackl if you're curious.